exp/etcdrestore/webhooks/rke2config.go

```diff
@@ -413,22 +451,15 @@ func (r *RKE2ConfigWebhook) createSystemAgentInstallScript(ctx context.Context,
 	installScriptKey := "install.sh"
 
 	serverUrlBash := fmt.Sprintf("CATTLE_SERVER=%s\n", serverUrl)
+	binaryURL := fmt.Sprintf("CATTLE_AGENT_BINARY_BASE_URL=\"%s/assets\"\n", serverUrl)
 
 	if err := r.Create(ctx, &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      installScriptSecretName,
 			Namespace: rke2Config.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
```

Comment:
any specific reason why we need to remove ownerref?

Reply:
Yes, we can't set the OwnerReference because at the moment when webhook is called RKE2Config doesn't exist and has no UID, which is required to be present. We will have to write a small controller to set owner references.